zebra/zebra_rib.c

```diff
@@ -3125,12 +3125,15 @@ void rib_sweep_table(struct route_table *table)
 	struct route_entry *re;
 	struct route_entry *next;
 	struct nexthop *nexthop;
+	int64_t startup_time = zrouter.startup;
 
 	if (!table)
 		return;
 
 	for (rn = route_top(table); rn; rn = srcdest_route_next(rn)) {
 		RNODE_FOREACH_RE_SAFE (rn, re, next) {
+			int64_t re_time = re->uptime;
```

Comment:
Isn't uptime in route_entry is of structure time_t type ?

Reply:
yeah I did this on purpose, as that I was planning on subtracting and didn't want to get overflow issues.  But maybe I need to think this through a bit more.  Fixed with a comparison operator.  Think things read a bit clearer now. thanks